### DISCUSSION TITLE
"Code Block" on GitHub Pages is not rendering and throws an error

### DISCUSSION THREAD

**Discussion by jharris1993** [community/community#23170] [New to GitHub]
<p>Situation:<br>
I am trying to teach myself the finer points of GitHub pages by cloning an existing documentation repository and then mess with it to see what happens.</p>
<p>The repository I decided to clone was the Raspberry Pi documentation repository located here:<br>
<a href="https://github.com/raspberrypi/documentation" class="inline-onebox" rel="noopener nofollow ugc">GitHub - raspberrypi/documentation: Official documentation for the Raspberry Pi</a></p>
<p>The file in question is the <strong>cmio-camera.md</strong> file.<br>
The original file is located here:<br>
<a href="https://github.com/raspberrypi/documentation/blob/6d7da8ebe337d4067a1e6fe53e637a12c3505ed5/hardware/computemodule/cmio-camera.md" class="inline-onebox" rel="noopener nofollow ugc">documentation/cmio-camera.md at 6d7da8ebe337d4067a1e6fe53e637a12c3505ed5 · raspberrypi/documentation · GitHub</a></p>
<p>My copy of it is located here:<br>
<a href="https://github.com/jharris1993/Test_Using_Raspberry_Pi_Documentation/blob/18f3a8ad1d5b9575cec6269d567469f32a02765f/hardware/computemodule/cmio-camera.md" class="inline-onebox" rel="noopener nofollow ugc">Test_Using_Raspberry_Pi_Documentation/cmio-camera.md at 18f3a8ad1d5b9575cec6269d567469f32a02765f · jharris1993/Test_Using_Raspberry_Pi_Documentation · GitHub</a></p>
<p>The problem I am experiencing:</p>
<p>When I try to create a GitHub page based on the material downloaded, I get the following error:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/5/c/5c3476d2655bf12af1e77845fe7a775ceb12517e.jpeg" data-download-href="/uploads/short-url/d9Gkx8aVW32HVHwBuWIyhRsNmKG.jpeg?dl=1" title="Your site is having problems building" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/12052596/181075422-3841622b-3ef3-45f7-894a-e1d17f53db5e.jpeg" alt="Your site is having problems building" data-base62-sha1="d9Gkx8aVW32HVHwBuWIyhRsNmKG" width="690" height="83" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/5/c/5c3476d2655bf12af1e77845fe7a775ceb12517e_2_690x83.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/5/c/5c3476d2655bf12af1e77845fe7a775ceb12517e_2_1035x124.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/3X/5/c/5c3476d2655bf12af1e77845fe7a775ceb12517e.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/5/c/5c3476d2655bf12af1e77845fe7a775ceb12517e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Your site is having problems building</span><span class="informations">1192×145 46.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>What I have done:</p>
<ol>
<li>
<p>The entire repo would not build since there was no index.md file.<br>
I copied the README.md file and renamed it index.md.</p>
</li>
<li>
<p>The modified repo then began to fail on cmia-camera.md with a particular line and variable not closed correctly.</p>
</li>
<li>
<p>Reading the documentation on formatting, it told me that code blocks should be <em><strong>indented by four spaces</strong></em> and made no mention of back-ticks.  It also said that anything inside the indented part would <em><strong>NOT</strong></em> be parsed, but would be presented <em>verbatim</em>.</p>
</li>
<li>
<p>I removed the back-ticks and replaced them with indenting the two lines four spaces.<br>
This also did not work.  I tried <code>[code]</code> and <code>[/code]</code> delimiters; they didn’t work either.</p>
</li>
</ol>
<p>This is the block of offending code.</p>
<pre><code class="lang-auto">```
MMAL_PARAMETER_INT32_T camera_num = {{MMAL_PARAMETER_CAMERA_NUM, sizeof(camera_num)}, CAMERA_NUMBER};
status = mmal_port_parameter_set(camera-&gt;control, &amp;camera_num.hdr);
```</code></pre>
<p>When I “view” the page, everywhere the three back-ticks surround a block, it’s properly rendered as a code block.  Obviously it works on the Raspberry Pi documentation pages, as it is up and published.</p>
<p>When I change the back-ticks to a four-space indent, the same thing happens - it formats correctly when “viewed” but throws the same error when I try to build the page.</p>
<p>If someone can tell me why this is not compiling correctly, I wold greatly appreciate it.</p>
<p>Thanks!</p>
<p>Jim “JR”</p>

**Comment by yamiacat** [comment, marked as answer]
<p>Hey <a class="mention" href="/jharris1993">@jharris1993</a>! :wave:</p>
<p>Hmmm, so, the <code>raspberrypi/documentation</code> repo you link to isn’t actually generating a GitHub Pages site, so they’re not using GitHub’s build system. How they process markdown during the page build process might be totally different.</p>
<p>On GitHub, at least, the code block you posted is causing problems because Jekyll’s parser treats any code using double curly braces {{ text }} or curly braces and percent signs {% text %} that’s included in inline or fenced code blocks as if it’s a Liquid tag.</p>
<p>To stop this happening you’ll need to surround any affected code blocks with some tags to tell Jekyll to ignore them:</p>
<hr>
<p>{% raw %}</p>
<p>your code here</p>
<p>{% endraw %}</p>
<hr>
<p>You can also surround these tags with Markdown comments, meaning they won’t render on <a href="http://GitHub.com">GitHub.com</a>:</p>
<hr>
<p>[//]: # ({% raw %})</p>
<p>your code here</p>
<p>[//]: # ({% endraw %})</p>
<hr>
<p>Hope this helps!</p>

**Comment by jharris1993** [comment]
<p>Perhaps this could be added to the documentation for code blocks since we can’t assume that any language is going to conform to Liquid’s standards.</p>
<p>Or perhaps an even better way would be to instruct Liquid that anything within a code block is automatically assumed to be “raw” which is the universally expected behavior of code blocks.</p>
<p>Thanks for your help, it’s appreciated!</p>

**Comment by jingydz** [comment]
“Unable to render code block”

https://github.com/PopicLab/cue/blob/master/notebooks/tutorial.ipynb

![image](https://github.com/community/community/assets/45711930/f0bdc162-a64f-4535-a5ba-321d6b217812)

**Reply by Eurekha-K** [reply]
I am getting the same error .
Could you tell me how did you resolve this?

**Reply by Eurekha-K** [reply]
From the past few days I have been trying to solve this .Help me with this error
